package.json

```diff
@@ -65,13 +65,13 @@
     "@types/jquery": "^3.5.1",
     "@types/lodash": "^4.14.158",
     "@types/node": "18.16.19",
-    "@types/react": "18.2.8",
-    "@types/react-dom": "18.2.4",
+    "@types/react": "18.2.13",
+    "@types/react-dom": "18.2.6",
     "@types/react-helmet": "6.1.6",
-    "@types/react-is": "18.2.0",
+    "@types/react-is": "18.2.1",
     "@types/styled-components": "5.1.26",
-    "@typescript-eslint/eslint-plugin": "5.59.8",
-    "@typescript-eslint/parser": "5.59.8",
+    "@typescript-eslint/eslint-plugin": "5.61.0",
+    "@typescript-eslint/parser": "5.61.0",
```

Comment:
Because `@nx/eslint-plugin:v16.5.0` [has a peer dependency](https://github.com/nrwl/nx/blob/16.5.0/packages/eslint-plugin/package.json#L27) `"@typescript-eslint/parser": "^5.60.1"` npm install fails unless you update it to `>= 5.60.1`, which you are doing with update to `v5.61.0`, which [brings](https://github.com/typescript-eslint/typescript-eslint/releases/tag/v5.61.0) us official Typescript 5.1 support.

I understand you should update `@typescript-eslint/eslint-plugin` to match the same version, but why did you update a bunch of `@types/*` packages for React?

Reply:
Since we are using `nx` I used `nx migrate` command to update `nx` plugins. So these updates are from automated migration.

Reply:
Ok, I only thought of migrations when doing major version upgrade of Nx, but you're right, using `nx migrate` is the correct way to upgrade. We've done plenty major version upgrades of Nx in the past (see #365 for latest one), and I don't like how `nx migrate` treats other dependencies. Sometimes, it can also downgrade our dependencies. In this case, I would keep the upgrade to React dependencies (I see that we can upgrade `@types/react` even more), but commit it separately from Nx upgrade.

Whether you manually change output of `nx migrate` (which I prefer) or not, could be a matter of "taste", but the actual problem is that you forgot to **run** migrations. `npx nx migrate latest` also creates `migrations.json` file (which should not be committed). After upgrading packages (in multiple commits like said), you should have run `npx nx migrate --run-migrations`. There is a minor change in `nx.json` caused by this migration.

Regarding commit messages, look at f62f3791af6a7ee359f7f1b8491098327fbeaf1e and efedfe9c787cb8a521b37e9b8979f9c009eba590 to see how I explained that commit was actually generated by Nx migration and in the second commit I even included console output from running a migration. In case of this update to 16.5.0, I've got this on my machine:

```
 >  NX   Running migrations from 'migrations.json'

Ran add-test-setup-to-inputs-ignore from @nx/jest
  Add test-setup.ts to ignored files in production input

  UPDATE nx.json
---------------------------------------------------------
```